src/TableStorage.Abstractions/Store/TableStore.cs

```diff
@@ -568,6 +568,53 @@ public async Task DeleteUsingWildcardEtagAsync(T record)
             await DeleteAsync(record).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Delete records by partition key
+        /// </summary>
+        /// <param name="partitionKey"></param>
+        public async Task DeleteByPartitionAsync(string partitionKey)
+        {
+            var deleteQuery = BuildGetByPartitionQuery(partitionKey);
+
+            TableContinuationToken continuationToken = null;
+            do
+            {
+                var tableQueryResult = CloudTable.ExecuteQuerySegmentedAsync(deleteQuery, continuationToken);
+
+                continuationToken = tableQueryResult.Result.ContinuationToken;
+
+                // Split result into chunks of 100s
+                List<List<T>> rowsChunked = tableQueryResult.Result.Select((x, index) => new { Index = index, Value = x })

```

Comment:
This library has a dependency on my Useful.Extensions which has a partition extension that would be useful here https://github.com/Tazmainiandevil/Useful.Extensions/blob/master/src/Useful.Extensions/EnumerableExtensions.cs

Reply:
There is an example in Insert that uses it and that it also handles different partition keys